packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -1,22 +1,43 @@
 import * as React from 'react';
 import { View } from 'react-native';
 import { ShadowProps, ShadowTokens, shadowName } from './Shadow.types';
-import { compressible, buildUseTokens, useFluentTheme, mergeProps } from '@fluentui-react-native/framework';
+import { compressible, buildUseTokens, useFluentTheme, mergeProps, UseTokens } from '@fluentui-react-native/framework';
 import { shadowStyleFromTheme } from './shadowStyle';
 
 const useShadowTokens = buildUseTokens<ShadowTokens>(() => ({}), shadowName);
 
-export const Shadow = compressible<ShadowProps, ShadowTokens>((props: ShadowProps) => {
+export const Shadow = compressible<ShadowProps, ShadowTokens>((props: ShadowProps, useTokens: UseTokens<ShadowTokens>) => {
   const { depth } = props;
   const theme = useFluentTheme();
 
+  const [tokens, cache] = useTokens(theme);
   const shadowStyle = shadowStyleFromTheme(theme, depth);
```

Comment:
I would move this line to inside the cache() call.

```js
const [keyShadowTokenStyle] = cache(
  () => {
    const shadowStyle = shadowStyleFromTheme(theme, depth);
    return {
      shadowColor: shadowStyle.key.shadowColor,
      shadowOpacity: shadowStyle.key.shadowOpacity,
      shadowRadius: shadowStyle.key.shadowRadius,
      shadowOffset: shadowStyle.key.shadowOffset,
    }
  },
  ['theme', 'depth'],
);
```

Reply:
I see, if it's outside the cache() call is it defeating the purpose of the cache?

Reply:
You also _can_ combine the two cache calls if you want, that's more of a stylistic choice. I think they both perform similar.

```js
const [shadowTokenStyleSet] = cache(
  () => (shadowStyleFromTheme(theme, depth), 
  ['theme', 'depth'],
);

....
<View style={shadowTokenStyleSet.ambient}>
  <View style={shadowTokenStyleSet.key}></View>
</View>
```
```

Reply:
> I see, if it's outside the cache() call is it defeating the purpose of the cache?

A bit. You're doing unnecessary work, but it won't cause an unnecessary render (since you're using the cached value in the end anyway)

Reply:
I see okay, thanks Saad!